src/Numerical/RootFinding/QuadraticEquation.cpp

```diff
@@ -0,0 +1,24 @@
+
+#include "Numerical/RootFinding/QuadraticEquation.hpp"
+
+#include <gsl/gsl_poly.h>
+
+#include "ErrorHandling/Assert.hpp"
+#include "ErrorHandling/ExpectsAndEnsures.hpp"
+
+double positive_root(const double a, const double b, const double c) {
+  double x0;
+  double x1;
+  gsl_poly_solve_quadratic(a, b, c, &x0, &x1);
+  Ensures(x0 <= 0.0 and x1 >= 0.0);
+  return x1;
```

Comment:
As long as the GSL doesn't cause the `signaling_nan` to trigger that's the best way to go. If it doesn't play nice (which as a safety precaution we probably don't want to risk) is then we write it as:
```cpp
if (gsl_poly_solve_quadratic(a, b, c, &x0, &x1)) {
  return /*signaling_nan()*/;
}
```
We should make sure that `std::isnan` works as desired with `signaling_nan`